omnigibson/utils/constants.py

```diff
@@ -178,17 +170,16 @@ class OccupancyGridState:
 
 
 @cache
-def semantic_class_name_to_id():
+def semantic_class_name_to_id(scene):
     """
     Get mapping from semantic class name to class id
 
     Returns:
         dict: class name to class id
     """
     categories = get_all_object_categories()
-    from omnigibson.systems.system_base import REGISTERED_SYSTEMS
 
-    systems = sorted(REGISTERED_SYSTEMS)
+    systems = sorted(scene.system_registry.object_names)
```

Comment:
how often is this call used? if it occurs every timestep during the vision sensor's get_obs() call, is there a way to make this more efficient, e.g.: cache the sorting somehow if the registry hasn't changed?

Reply:
This only happens once